lib/gollum-lib/helpers.rb

```diff
@@ -1,13 +1,23 @@
 # ~*~ encoding: utf-8 ~*~
+
 module Gollum
   module Helpers
-
-    def trim_leading_slash(url)
-      return url if url.nil?
+    
+    # If url starts with a leading slash, trim down its number of leading slashes to 1. Else, return url unchanged.
+    def trim_leading_slashes(url)
+      return nil if url.nil?
       url.gsub!('%2F', '/')
       return '/' + url.gsub(/^\/+/, '') if url[0, 1] == '/'
       url
     end
+    
+    # Take a path and turn it into a string for display as link text.
+    # For example:
+    # '/opt/local/bin/ruby.ext' -> 'ruby'
+    def path_to_link_text(str, is_path = true)
```

Comment:
What's a little confusing is that the comment states that this methods takes a path, but the method signature says it takes a String `str` that may or may not be a path.

Reply:
Good point! I can just change the comment, unless you see a better way of implementing the logic. See my comment in `tags.rb`

Reply:
And what about changing `is_path` to `for_page`? You are passing in `!!page` after all.

Reply:
👍